packages/web/src/common/store/cache/tracks/utils/fetchTrackStreamUrls.ts

```diff
@@ -30,7 +36,7 @@ export function* fetchTrackStreamUrls({ trackIds }: { trackIds: ID[] }) {
       try {
         const existingUrl = yield* select(getTrackStreamUrl, id)
 
-        if (existingUrl) {
+        if (existingUrl && !isUpdate) {
           return { [id]: existingUrl }
```

Comment:
Adding the update flag option is a bit of a workaround, but it still seems correct to me. 
I added logic to dedupe cache adds since each track add results in a network call now. The dedupe logic to trim network calls feels important and this is the only case I can think of where it backfired, so I think this is the correct move.

Reply:
got it, thank you for clarification